src/server.jl

```diff
@@ -40,9 +40,6 @@ mutable struct ServerOptions
     tlsconfig::HTTP.TLS.SSLConfig
     readtimeout::Float64
     ratelimit::Rational{Int}
-    maxuri::Int64
-    maxheader::Int64
-    maxbody::Int64
```

Comment:
why remove these?

Reply:
I could not see any place where they were used.
I assumed that these were a hang-over from original C code where there was some kind of limited size buffer.
It seems better not to impose arbitrary limits on data size unless there is a specific reason to.

Reply:
On the contrary, these limits are to _protect_ a server from potential malicious attacks. They may have not been fully hooked up, but the idea is that you can configure a server to only accept URIs/headers/body of certain sizes before throwing an error (instead of, for example, running out of memory on a device and totally failing). I think it's important to keep these in.

Reply:
OK, fair enough, I think this should be doable as a seperate optional layer rather than being done at the parser level. See #141

Reply:
Hmmmm, I'm not sure why we wouldn't do this at the parser level? If we allow the parser to parse some malicious request that has 20GB headers, then I think we're already lost by the time it gets to the server layer, right? Was the parsing code that already enforced these overflow limits removed in your refactoring?

Reply:
Yes, that code was one of the very first things I removed.
What if the parser is the code that is susceptible to large input attacks?
The check for large input should be before the parser, not after.
e.g. maybe the loop in `Messages.readheaders()` should count bytes sent to the parser and throw an error if the parser has not found the end of the headers after 100k?:
https://github.com/samoconnor/HTTP.jl/blob/a8f4f730b8b203327efc8e12aed731e7b324aead/src/Messages.jl#L400